BitList.cs

```diff
@@ -0,0 +1,1553 @@
+                    c = 0;
+                }
+                bytesList[bytesList.Count - 1][c] = bytes[i];
+            }
+            var floatList = new List<float>();
+            foreach (var fbytes in bytesList)
+                floatList.Add(BitConverter.ToSingle(fbytes, 0));
+            return floatList.ToArray();
+        }
+    }
+
+    public double[] ToDoubleArray()
+    {
+        if (IsEmpty)
+            return new double[0];
+        else
+        {
+            var bytes = ToByteArray();
+            var bytesList = new List<byte[]>();
+            var c = -1;
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                c++;
+                if (i == 0 | c == 8)
+                {
+                    bytesList.Add(new byte[8]);
+                    c = 0;
+                }
+                bytesList[bytesList.Count - 1][c] = bytes[i];
+            }
+            var doubleList = new List<double>();
+            foreach (var dbytes in bytesList)
+                doubleList.Add(BitConverter.ToDouble(dbytes, 0));
+            return doubleList.ToArray();
+        }
+    }
+
+    public decimal[] ToDecimalArray()
+    {
+        if (IsEmpty)
+            return new decimal[0];
+        else
+        {
+            try
+            {
+                var ints = ToInt32Array();
+                var intsList = new List<int[]>();
+                var c = -1;
+                for (int i = 0; i < ints.Length; i++)
+                {
+                    c++;
+                    if (i == 0 | c == 4)
+                    {
+                        intsList.Add(new int[4]);
+                        c = 0;
+                    }
+                    intsList[intsList.Count - 1][c] = ints[i];
+                }
+                var decimalList = new List<decimal>();
+                foreach (var dints in intsList)
+                    decimalList.Add(new decimal(dints));
+                return decimalList.ToArray();
+            }
+            catch
+            {
+                throw new ConversionErrorException();
+            }
+        }
+    }
+
+    public char[] ToCharArray()
+    {
+        if (IsEmpty)
+            return new char[0];
+        else
+        {
+            var bytes = ToByteArray();
+            var bytesList = new List<byte[]>();
+            var c = -1;
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                c++;
+                if (i == 0 | c == 4)
+                {
+                    bytesList.Add(new byte[4]);
+                    c = 0;
+                }
+                bytesList[bytesList.Count - 1][c] = bytes[i];
+            }
+            var charList = new List<char>();
+            foreach (var cbytes in bytesList)
+                charList.Add(BitConverter.ToChar(cbytes, 0));
+            return charList.ToArray();
+        }
+    }
+
+    public override string ToString()
+    {
+        if (IsEmpty)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)